### ISSUE TITLE
False positive E0594 in for loop.

### ISSUE THREAD

**Issue by ralfbiedert** [intellij-rust/intellij-rust#3782]
## Environment

* **Intellij-Rust plugin version:** 0.2.97.2123-191
* **Rust toolchain version:**  rustc 1.36.0-nightly
* **IDE name and version:** Clion 2019.01
* **Operating system:** macOS 10.14.4

## Problem description

The Rust plugin gives a false positive E0594 on the `x.f = 1`; assignment in this code:

```rust

#[derive(Default, Copy, Clone)]
struct S {
    f: u32
}

fn main() {
    let mut v = [S::default(); 5];
    
    for x in &mut v {
        x.f = 1;
    }
}
```
![Screen Shot 2019-05-03 at 14 15 44](https://user-images.githubusercontent.com/301288/57136717-fb09bd00-6dad-11e9-9933-6cab3440d71b.png)

However, compiling this code with Rust works just fine:

```
/Users/rb/.cargo/bin/cargo run --color=always --release
    Finished release [optimized] target(s) in 0.02s
     Running `target/release/mapgen`

Process finished with exit code 0
```
